### ISSUE TITLE
Instagram OAuth Provider - Authentication issue

### ISSUE THREAD

**Issue by FilipCotan** [aspnet-contrib/AspNet.Security.OAuth.Providers#227]
Hi there,

I'm trying to authenticate with Instagram as an external login provider in a .Net Core 2.0 application.
My middleware configuration looks like this: 
Startup.cs
```csharp
services.AddAuthentication().AddInstagram(options =>
{
  options.ClientId = ApplicationSettings.InstagramSettings.ApplicationId;
  options.ClientSecret = ApplicationSettings.InstagramSettings.ApplicationSecret;
  options.Scope.Add(ApplicationSettings.InstagramSettings.Permissions);
  options.SaveTokens = true;
});
```

ExternalLogin method in AccountController is unchanged:

```csharp
        [HttpPost]
        [AllowAnonymous]
        [ValidateAntiForgeryToken]
        public IActionResult ExternalLogin(string provider, string returnUrl = null)
        {
            // Request a redirect to the external login provider.
            var redirectUrl = Url.Action(nameof(ExternalLoginCallback), "Account", new { returnUrl });
            var properties = _signInManager.ConfigureExternalAuthenticationProperties(provider, redirectUrl);

            return Challenge(properties, provider);
        }
```

Yet, when 
```csharp
var info = await _signInManager.GetExternalLoginInfoAsync();
```
 is called, I get null as result. 

NOTE: 
- This works perfectly fine for Facebook provider, with no other settings in the middleware configuration.
- If I inspect the network activity in Fiddler, I can see that the Instagram successfully sends the access_token back to my application, only that GetExternalLoginInfoAsync() can't read it. 

Any ideas?

Cheers,
Filip

**Comment by kevinchalet** [comment]
Hey.

Have you tried using the provider without ASP.NET Core Identity? (i.e like we do in the MVC sample)

**Comment by FilipCotan** [comment]
No, I haven't. I'll try it tho and I'll let you know. However, I had to handle `OnTicketReceived` event in Instagram configuration and save the `access_token` in Session, add a new `ExternalLoginCallback` method for Instagram and grab the `access_token` from Session ( and then delete it :D ) and save it in the database. It's more like a workaround, but it works if this helps anyone.

**Comment by kevinchalet** [comment]
Question: what's the exact version you're using?

**Comment by FilipCotan** [comment]
I'm using version: 2.0.0-rc2-final.

**Comment by kevinchalet** [comment]
There's a bug in the RC2 final version that prevents the userinfo response from being correctly processed (which means the `ClaimType.NameIdentifier` claim required by Identity is not added).

Give the RTM nightly builds a try to see if it fixes your issue.

**Comment by FilipCotan** [comment]
It looks like the bug was fixed in the latest RTM nightly build (2.0.0-rtm-0580) and it successfully returns the external login information. Thanks for your help, @PinpointTownes . Cheers.
